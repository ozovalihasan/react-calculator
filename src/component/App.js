import React, { useState } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';
import '../index.scss';

const App = () => {
  const [total, setTotal] = useState('');
  const [next, setNext] = useState('');
  const [operation, setOperation] = useState('');

  const handleClick = buttonName => {
    const result = calculate({ next, total, operation }, buttonName);
    setNext(result.next);
    setTotal(result.total);
    setOperation(result.operation);
  };

  return (
    <div className="app">
      <Display next={next} total={total} operation={operation} />
      <ButtonPanel clickHandler={handleClick} />
    </div>
  );
};
export default App;
