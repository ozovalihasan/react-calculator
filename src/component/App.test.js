import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import App from './App';

describe('<App />', () => {
  it('shows static text', () => {
    render(<App />);
    expect(screen.getByText("Let's do some math")).toBeTruthy();
  });

  it('shows result of operations done correctly', () => {
    render(<App />);
    fireEvent.click(screen.getByText('7'));
    fireEvent.click(screen.getByText('+'));
    fireEvent.click(screen.getByText('8'));
    fireEvent.click(screen.getByText('-'));
    expect(screen.getByText(/15-/)).toBeTruthy();

    fireEvent.click(screen.getByText('3'));
    fireEvent.click(screen.getByText('='));
    expect(screen.getByText(/12/)).toBeTruthy();
  });
});
