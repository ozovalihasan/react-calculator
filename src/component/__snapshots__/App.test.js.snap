// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<div
  className="app main"
>
  <div
    className="app title"
  >
    Let's do some math
  </div>
  <div
    className="app calculator"
  >
    <div
      className="display main"
    >
      <div
        className="display w-100"
      >
        0
      </div>
    </div>
    <div
      className="button-panel main"
    >
      <button
        className="button main w-25"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "darkgray",
          }
        }
        type="button"
      >
        AC
      </button>
      <button
        className="button main w-25"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "darkgray",
          }
        }
        type="button"
      >
        +/-
      </button>
      <button
        className="button main w-25"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "darkgray",
          }
        }
        type="button"
      >
        %
      </button>
      <button
        className="button main w-25"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="button-panel main"
    >
      <button
        className="button main w-25"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "darkgray",
          }
        }
        type="button"
      >
        7
      </button>
      <button
        className="button main w-25"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "darkgray",
          }
        }
        type="button"
      >
        8
      </button>
      <button
        className="button main w-25"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "darkgray",
          }
        }
        type="button"
      >
        9
      </button>
      <button
        className="button main w-25"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        X
      </button>
    </div>
    <div
      className="button-panel main"
    >
      <button
        className="button main w-25"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "darkgray",
          }
        }
        type="button"
      >
        4
      </button>
      <button
        className="button main w-25"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "darkgray",
          }
        }
        type="button"
      >
        5
      </button>
      <button
        className="button main w-25"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "darkgray",
          }
        }
        type="button"
      >
        6
      </button>
      <button
        className="button main w-25"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="button-panel main"
    >
      <button
        className="button main w-25"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "darkgray",
          }
        }
        type="button"
      >
        1
      </button>
      <button
        className="button main w-25"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "darkgray",
          }
        }
        type="button"
      >
        2
      </button>
      <button
        className="button main w-25"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "darkgray",
          }
        }
        type="button"
      >
        3
      </button>
      <button
        className="button main w-25"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="button-panel main"
    >
      <button
        className="button main w-50"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "darkgray",
          }
        }
        type="button"
      >
        0
      </button>
      <button
        className="button main w-25"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "darkgray",
          }
        }
        type="button"
      >
        .
      </button>
      <button
        className="button main w-25"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "orange",
          }
        }
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
